src/javaMain/kotlin/io/github/oshai/kotlinlogging/internal/KLoggerNameResolver.kt

```diff
@@ -1,41 +1,28 @@
 package io.github.oshai.kotlinlogging.internal
 
-import java.lang.reflect.Modifier
+import kotlin.reflect.KClass
 
 /** Resolves name of java classes */
 internal actual object KLoggerNameResolver {
 
   /** get class name for function by the package of the function */
   internal actual fun name(func: () -> Unit): String {
-    val name = func.javaClass.name
-    val slicedName =
-      when {
-        name.contains("Kt$") -> name.substringBefore("Kt$")
-        name.contains("$") -> name.substringBefore("$")
-        else -> name
-      }
-    return slicedName
+    return name(func::class)
+  }
+
+  internal fun name(clazz: KClass<*>): String {
```

Comment:
I think this method should be private

Reply:
@oshai historically this method was used in the tests, therefore it's not possible to make it private without rewriting tests structure. https://github.com/oshai/kotlin-logging/blob/master/src/javaMain/kotlin/io/github/oshai/kotlinlogging/internal/KLoggerNameResolver.kt#L21

I would prefer to keep it internal for now and if the changes from https://github.com/oshai/kotlin-logging/pull/481 will be accepted, then signature of the method will be changed from
```kotlin
internal actual fun name(func: () -> Unit): String
```

to
```kotlin
internal actual fun name(ref: Any): String {
```

and then we can update the tests to use instances of the classes, rather then KClass